src/shared/extensionGlobals.ts

```diff
@@ -27,6 +27,22 @@ export namespace ext {
     export let templateRegistry: CloudFormationTemplateRegistry
     export let codelensRootRegistry: CodelensRootRegistry
 
+    let _didReload = false
+
+    export function init(context: ExtensionContext, window: Window) {
```

Comment:
Nit: add a safeguard to ensure this can only be called once?

Reply:
It's more of an interface to replace the current shape where we manually set the globals (which also aren't guarded). Also tests need to reset these sometimes.